src/tunnel.c

```diff
@@ -425,7 +429,7 @@ static int tcp_connect(struct tunnel *tunnel)
 	log_debug("gateway_port: %u\n", tunnel->config->gateway_port);
 
 	ret = connect(handle, (struct sockaddr *) &server, sizeof(server));
-	if (ret == -1) {
```

Comment:
I was checking explicitly for the [failure value -1](http://pubs.opengroup.org/onlinepubs/009695399/functions/connect.html), but your code is more tolerant about different implementations of the connect() function